src/Arcus.Scripting.Storage.Table/Scripts/Set-AzTableStorageEntities.ps1

```diff
@@ -0,0 +1,71 @@
+Write-Verbose "Retrieving Azure storage account context for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'..."
+$storageAccount = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+if ($storageAccount -eq $null)
+{
+    throw "Retrieving Azure storage account context for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName' failed."
+}
+$ctx = $storageAccount.Context
+Write-Verbose "Azure storage account context has been retrieved for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'"
+
+# Retrieve Azure storage table
+Write-Verbose "Retrieving Azure storage table '$TableName' for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'..."
+$storageTable = Get-AzStorageTable -Name $TableName -Context $ctx
+if ($storageTable -eq $null)
+{
+    throw "Retrieving Azure storage table '$TableName' for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName' failed."
+}
+$cloudTable = ($storageTable).CloudTable
+Write-Verbose "Azure storage table '$TableName' has been retrieved for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'"
+
+# Delete all existing entities 
+Write-Host "Deleting all existing entities in Azure storage table '$TableName' for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'..."
+$entitiesToDelete = Get-AzTableRow -table $cloudTable
+$deletedEntities = $entitiesToDelete | Remove-AzTableRow -table $cloudTable
+Write-Host "Successfully deleted all existing entities in Azure storage table '$TableName' for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'"
+
+# Create all new entities specified in json file
+$configFile = Get-Content -Path $ConfigurationFile | ConvertFrom-Json
+foreach ($entityToAdd in $configFile) {
+    # Check if PartitionKey provided
+    if ($entityToAdd.PartitionKey -ne $null) {
+        $partitionKey = $entityToAdd.PartitionKey
+        $entityToAdd.PSObject.Properties.Remove('PartitionKey')
+    } else {
+        $partitionKey = New-Guid
+    }
+
+    # Check if RowKey provided
+    if ($entityToAdd.RowKey -ne $null) {
+        $rowKey = $entityToAdd.RowKey
+        $entityToAdd.PSObject.Properties.Remove('RowKey')
+    } else {
+        $rowKey = New-Guid
+    }
+
+    # Convert psObject to hashtable
+    $entityHash=@{}
+    $entityToAdd.PSObject.Properties | Foreach { $entityHash[$_.Name] = $_.Value }
+
+    # Create entity in table storage
+    $addedRow = Add-AzTableRow `
+    -table $cloudTable `
+    -partitionKey $partitionKey `
+    -rowKey $rowKey `
+    -property $entityHash
+
+    Write-Verbose "Successfully added row with PartitionKey '$partitionKey' and RowKey '$rowKey' to Azure storage table '$TableName' for Azure storage account '$StorageAccountName' in resource group '$ResourceGroupName'"
```

Comment:
Is it 'added' when the record already existed? Maybe '(re)added' is better suited here?

Reply:
Yeah the record could have existed, but it was deleted and then added again.

The logs show it is first being deleted and then added, so for me it makes sense the way it is right now.
Also if you do not specify `PartitionKey` and/or `RowKey` you are not really re-adding exactly re-adding the same data.